src/extensions/default/CodeFolding/Prefs.js

```diff
@@ -85,8 +93,9 @@ define(function (require, exports, module) {
       * @param {Object} folds the fold ranges to save for the current document
       */
     function setFolds(path, folds) {
-        store[path] = simplify(folds);
-        prefs.set(foldsKey, store);
+        var allFolds = PreferencesManager.getViewState(foldsKey);
```

Comment:
@MarcelGerber Yep, good point -- that should remove the need for `|| {}` both here & in `getFolds()`.